core/pkg/hostsensorutils/hostsensordeploy.go

```diff
@@ -92,6 +93,31 @@ func (hsh *HostSensorHandler) Init(ctx context.Context) error {
 	return nil
 }
 
+// checkNamespaceWasPresent check if the given namespace was already present on kubernetes.
+// Return true in case it find the namespace on the list, false otherwise.
+// In case we have some error with the kubernetes APIs, it returns an error.
+func (hsh *HostSensorHandler) checkNamespaceWasPresent(namespace string) (bool, error) {
+	namespaceList, err := hsh.k8sObj.KubernetesClient.
+		CoreV1().
+		Namespaces().
+		List(hsh.k8sObj.Context, metav1.ListOptions{})
```

Comment:
[Question]: Why not Get the namespace (based on name), if found - return true, otherwise, return false.

Reply:
Sorry, my fault. I'm going to change it.